packages/ui-components/src/components/windowedlist.ts

```diff
@@ -1260,10 +1331,118 @@ export class WindowedLayout extends PanelLayout {
  * A namespace for windowed list
  */
 export namespace WindowedList {
+  /**
+   * The default renderer class for windowed lists.
+   */
+  export class Renderer<T = any> implements IRenderer<T> {
+    /**
+     * Create the outer, root element of the windowed list.
+     */
+    createOuter(): HTMLElement {
+      return document.createElement('div');
+    }
+
+    /**
+     * Create the virtual scrollbar element.
+     */
+    createScrollbar(): HTMLUListElement {
+      return document.createElement('ul');
+    }
+
+    /**
+     * Create an individual item rendered in the scrollbar.
+     */
+    createScrollbarItem(index: number, item: T): HTMLLIElement {
+      const li = document.createElement('li');
+      li.appendChild(document.createTextNode(`${index}`));
+      return li;
+    }
+
+    /**
+     * Create the viewport element into which virtualized children are added.
+     */
+    createViewport(): HTMLElement {
+      return document.createElement('div');
+    }
+  }
+
+  export class Scrollbar<T = any> extends Widget {
+    constructor(model: IModel<T>, renderer: IRenderer<T>) {
+      super();
+      this.model = model;
+      this.renderer = renderer;
+      this.content = this.node.appendChild(renderer.createScrollbar());
+      this.content.className = 'jp-WindowedListScrollbar-content';
+      this.addClass('jp-WindowedListScrollbar');
+    }
+
+    get scrolled(): ISignal<this, number> {
+      return this._scrolled;
+    }
+
+    handleEvent(event: Event): void {
+      switch (event.type) {
+        case 'pointerdown':
+          this._evtPointerDown(event as PointerEvent);
+          break;
+      }
+    }
+
+    protected onAfterAttach(msg: Message): void {
+      super.onAfterAttach(msg);
+      this.node.addEventListener('pointerdown', this);
+    }
+
+    protected onBeforeAttach(msg: Message): void {
+      this.node.removeEventListener('pointerdown', this);
+      super.onBeforeDetach(msg);
+    }
+
+    protected onUpdateRequest(msg: Message): void {
+      const { content, renderer } = this;
+
+      while (content.firstChild) {
+        content.removeChild(content.firstChild);
+      }
+
+      const list = this.model.itemsList;
+      if (list?.length) {
+        for (let index = 0; index < list.length; index += 1) {
+          const item = renderer.createScrollbarItem(index, list.get(index));
+          item.dataset.index = `${index}`;
+          content.appendChild(item);
+        }
+      }
+    }
+
+    protected content: HTMLUListElement;
+    protected model: IModel<T>;
+    protected renderer: IRenderer<T>;
+
+    private _evtPointerDown(event: PointerEvent): void {
```

Comment:
What should happen if the user move the pointer when keeping it pressed? Should we listen for pointer motion and/or pointer up too?

Reply:
I am treating this like simple clicks for the initial implementation. I wonder if that is sufficient.